MapboxAndroidDemo/src/main/java/com/mapbox/mapboxandroiddemo/examples/extrusions/RotationExtrusionActivity.java

```diff
@@ -0,0 +1,186 @@
+    // object or in the same activity which contains the mapview.
+    Mapbox.getInstance(this, getString(R.string.access_token));
+
+    // This contains the MapView in XML and needs to be called after the access token is configured.
+    setContentView(R.layout.activity_extrusion_rotation);
+
+    mapView = (MapView) findViewById(R.id.mapView);
+    mapView.onCreate(savedInstanceState);
+    mapView.getMapAsync(new OnMapReadyCallback() {
+        @Override
+        public void onMapReady(@NonNull final MapboxMap map) {
+          mapboxMap = map;
+          setupBuildings();
+        }
+    });
+
+    //initialize sensors
+    sensorManager = (SensorManager) getSystemService(Context.SENSOR_SERVICE);
+
+    gyro = sensorManager.getDefaultSensor(Sensor.TYPE_ACCELEROMETER);
+    magnetic = sensorManager.getDefaultSensor(Sensor.TYPE_MAGNETIC_FIELD);
+
+    sensorManager.registerListener(this, gyro , 200);
+    sensorManager.registerListener(this, magnetic , 200);
+  }
+
+  private void setupBuildings() {
+    FillExtrusionLayer fillExtrusionLayer = new FillExtrusionLayer("3d-buildings", "composite");
+    fillExtrusionLayer.setSourceLayer("building");
+    fillExtrusionLayer.setFilter(eq("extrude", "true"));
+    fillExtrusionLayer.setMinZoom(15);
+    fillExtrusionLayer.setProperties(
+            fillExtrusionColor(Color.LTGRAY),
+            fillExtrusionHeight(Function.property("height", new IdentityStops<Float>())),
+            fillExtrusionBase(Function.property("min_height", new IdentityStops<Float>())),
+            fillExtrusionOpacity(0.9f)
+    );
+    mapboxMap.addLayer(fillExtrusionLayer);
+  }
+
+  @Override
+  public boolean onCreateOptionsMenu(Menu menu) {
+    return true;
+  }
+
+  @Override
+  public boolean onOptionsItemSelected(MenuItem item) {
+    return true;
+  }
+
+  @Override
+  protected void onStart() {
+    super.onStart();
+    mapView.onStart();
+  }
+
+  @Override
+  protected void onResume() {
+    super.onResume();
+    mapView.onResume();
+  }
+
+  @Override
+  protected void onPause() {
+    super.onPause();
+    mapView.onPause();
+  }
+
+  @Override
+  protected void onStop() {
+    super.onStop();
+    mapView.onStop();
+  }
+
+  @Override
+  public void onSaveInstanceState(Bundle outState) {
+    super.onSaveInstanceState(outState);
+    mapView.onSaveInstanceState(outState);
+  }
+
+  @Override
+  public void onLowMemory() {
+    super.onLowMemory();
+    mapView.onLowMemory();
+  }
+
+  @Override
+  public void onDestroy() {
+    super.onDestroy();
+    mapView.onDestroy();
+  }
+
+  @Override
+  public void onSensorChanged(SensorEvent event) {
+
+    if (event.sensor.getType() == Sensor.TYPE_ACCELEROMETER) {
+      gravArray = event.values;
+    }
+
+    if (event.sensor.getType() == Sensor.TYPE_MAGNETIC_FIELD) {
+      magneticArray = event.values;
+    }
+
+    if (gravArray != null && magneticArray != null) {
+      float[] num1 = new float[9];
+      float[] num2 = new float[9];
```

Comment:
Additionally `num1` and `num2` aren't great names for variables, they tell the reader nothing about the context. What they are or what they represent.